### ISSUE TITLE
Open www links without protocol

### ISSUE THREAD

**Issue by GoogleCodeExporter** [mintty/mintty#345]
```
What steps will reproduce the problem?
1. Right-click a link that starts with www, not http
2. Click open (first option)


What is the expected output? What do you see instead?
I expect to see it opened in a browser client, like it works with the http 
protocol.
What I get is a ~20 second halt and this error message:
"A device attached to the system is not functioning."


What versions of mintty, Cygwin/MSYS, and Windows are you using?
mintty 1.1.1, Cygwin 1.7.15-1, Windows 7 64bit


Please provide any additional information below.
This occured with mintty 1.0.3 as well, but with a different error text.
```

Original issue reported on code.google.com by `adrianlund` on 10 Jul 2012 at 5:17

**Comment by rgburke** [comment]
A simple change to handle www addresses is:

```diff
diff --git a/src/winclip.c b/src/winclip.c
index 02b7785..137e184 100644
--- a/src/winclip.c
+++ b/src/winclip.c
@@ -52,7 +52,7 @@ win_open(wstring wpath)
   wstring p = wpath;
   while (iswalpha(*p)) p++;

-  if (*wpath == '\\' || *p == ':') {
+  if (*wpath == '\\' || *p == ':' || (wcslen(wpath) > 4 && wcsncmp(L"www.", wpath, 4) == 0)) {
     // Looks like it's a Windows path or URI
     shell_exec(wpath);
   }
```

This works for me when selecting www.google.co.uk and clicking Open.

A potential issue with this approach is that it would match files beggining with `www.` and would also miss addresses in upper case (e.g. WWW.GOOGLE.CO.UK). I don't know how likely it is that anyone would encounter these cases during normal use though.

Any thoughts on this?

**Comment by mintty** [comment]
Included in 2.2.4, thank you.
